actions/create_handler.go

```diff
@@ -0,0 +1,118 @@
+package actions
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/castai/cluster-controller/castai"
+)
+
+type createHandler struct {
+	log    logrus.FieldLogger
+	client dynamic.Interface
+}
+
+func newCreateHandler(log logrus.FieldLogger, client dynamic.Interface) ActionHandler {
+	return &createHandler{
+		log:    log,
+		client: client,
+	}
+}
+
+func (h *createHandler) Handle(ctx context.Context, action *castai.ClusterAction) error {
+	req, ok := action.Data().(*castai.ActionCreate)
+	if !ok {
+		return newUnexpectedTypeErr(action.Data(), req)
+	}
+
+	if req.Object == nil {
+		return errors.New("no object provided")
+	}
+
+	newObj := &unstructured.Unstructured{Object: req.Object}
+	if newObj.GetNamespace() == "" {
+		return errors.New("object namespace is missing")
+	}
+
+	log := h.log.WithFields(logrus.Fields{
+		"id":       action.ID,
+		"action":   reflect.TypeOf(action.Data()).String(),
+		"group":    req.Group,
+		"resource": req.Resource,
+		"version":  req.Version,
```

Comment:
I would construct proper GVR resource as single field. Easier to read in logs...

Reply:
makes sense, had similar idea initially but kept it consistent with `patch` handler. Will change it there as well 🙏